nilearn/regions/tests/test_hierarchical_kmeans_clustering.py

```diff
@@ -6,40 +6,35 @@
 from nilearn._utils import compare_version
 from nilearn._utils.class_inspect import check_estimator
 from nilearn._utils.data_gen import generate_fake_fmri
-from nilearn.input_data import NiftiMasker
-from nilearn.maskers import SurfaceMasker
+from nilearn.maskers import NiftiMasker, SurfaceMasker
 from nilearn.regions.hierarchical_kmeans_clustering import (
     HierarchicalKMeans,
     _adjust_small_clusters,
     hierarchical_k_means,
 )
+from nilearn.surface import SurfaceImage
+from nilearn.surface.tests.test_surface import flat_mesh
 
 extra_valid_checks = [
     "check_clusterer_compute_labels_predict",
-    "check_clustering",
     "check_complex_data",
-    "check_dict_unchanged",
     "check_do_not_raise_errors_in_init_or_set_params",
     "check_dont_overwrite_parameters",
     "check_dtype_object",
     "check_estimator_sparse_array",
     "check_estimator_sparse_matrix",
-    "check_estimators_dtypes",
     "check_estimators_empty_data_messages",
-    "check_estimators_fit_returns_self",
-    "check_estimators_pickle",
     "check_estimators_unfitted",
     "check_f_contiguous_array_estimator",
     "check_fit2d_1sample",
     "check_fit2d_1feature",
-    "check_fit_check_is_fitted",
     "check_fit1d",
-    "check_fit_score_takes_y",
     "check_no_attributes_set_in_init",
-    "check_pipeline_consistency",
-    "check_readonly_memmap_input",
     "check_transformers_unfitted",
     "check_transformer_n_iter",
+    "check_methods_subset_invariance",
+    "check_methods_sample_order_invariance",
+    "check_dict_unchanged",
```

Comment:
Hmm... This one passes with old dependencies but fails with new ones. Any pointers @Remi-Gau ? Otherwise I could just do the sklearn-version-check thing

Reply:
`check_dict_unchanged` I meant

Reply:
Sklearn version check sounds appropriate then